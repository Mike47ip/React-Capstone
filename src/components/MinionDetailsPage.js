/* eslint-disable indent */
import React from 'react';
import { useParams, Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { Helmet } from 'react-helmet';
import '../styles/detailsPage.css';

function MinionDetailsPage() {
  const { minionId } = useParams();
  const selectedMinion = useSelector((state) =>
   state.minions.minions.find((minion) => minion.id === parseInt(minionId))
  );

  if (!selectedMinion) {
    return <div>Minion not found</div>;
  }

  return (
    <div className="minion-details-container">
      <Helmet>
        <link
          rel="stylesheet"
          href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@20..48,100..700,0..1,-50..200"
        />
      </Helmet>
      <article className="minion-details">
        {/* Use the Material Icons for the arrow back */}
        <Link to="/" className="material-symbols-outlined go-back-icon">
          <span className="material-symbols-outlined">arrow_back_ios</span>
        </Link>
        <h1>{selectedMinion.name}</h1>
        <img src={selectedMinion.image} alt={selectedMinion.name} />
        <span>
          <strong>Description:</strong>
          {' '}
          {selectedMinion.description}
        </span>
        <span>
          <strong>Tooltip:</strong>
          {' '}
          {selectedMinion.tooltip}
        </span>
        <span>
          <strong>Tradeable:</strong>
          {' '}
          {selectedMinion.tradeable ? 'Yes' : 'No'}
        </span>
      </article>
    </div>
  );
}

export default MinionDetailsPage;
