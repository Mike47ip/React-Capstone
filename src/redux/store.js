// src/store.js
import { configureStore } from '@reduxjs/toolkit';
import minionsReducer from './features/minionsSlice';

const store = configureStore({
  reducer: {
    minions: minionsReducer,
  },
});

export default store;
