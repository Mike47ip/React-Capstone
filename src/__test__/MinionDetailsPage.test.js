import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import MinionDetailsPage from '../components/MinionDetailsPage';

describe('Test if details page renders correctly', () => {
  test('Test if details page renders correctly', () => {
    const Tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <MinionDetailsPage />
        </BrowserRouter>
      </Provider>,
    );
    expect(Tree).toMatchSnapshot();
  });
});
