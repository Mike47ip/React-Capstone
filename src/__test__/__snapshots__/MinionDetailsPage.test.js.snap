// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if details page renders correctly Test if details page renders correctly 1`] = `
<div>
  Minion not found
</div>
`;

exports[`Test if details page renders correctly Test if navBar renders correctly 1`] = `
<div>
  Minion not found
</div>
`;

exports[`Test if details renders correctly Test if navBar renders correctly 1`] = `
<div>
  Minion not found
</div>
`;

exports[`Test if navBar renders correctly Test if navBar renders correctly 1`] = `
<div>
  Minion not found
</div>
`;
